core/src/Kokkos_CopyViews.hpp

```diff
@@ -830,14 +830,36 @@ struct ZeroMemset {
   }
 };
 
+// Returns true when we can safely determine that the object has all 0 bits,
+// false otherwise.  It is intended to determine whether to perform zero memset
+// as an optimization.
+template <typename T>
+bool has_all_zero_bits(const T& value) {
+  static_assert(std::is_trivial_v<T>);
+
+  if constexpr (std::is_scalar_v<T>) {
+    return value == T();
+  }
+
+  KOKKOS_IMPL_DISABLE_UNREACHABLE_WARNINGS_PUSH()
+  if constexpr (std::is_standard_layout_v<T> &&
```

Comment:
Why do we need `is_standard_layout_v` ?

Reply:
The Clang-Tidy check documentation pointed out by Nic in https://github.com/kokkos/kokkos/pull/7980#pullrequestreview-2754618456 clearly calls out that requirement and includes a reference.